### DISCUSSION TITLE
Proposal 1: Draft PRs and Ready for Review

### DISCUSSION THREAD

**Discussion by tomhealey-icma** [finos/common-domain-model#2789] [Meetings]
Current: Most PRs are submitted as “review ready” even if they are not labeled.
Proposal: All PRs will be submitted as Draft. A contributor/editor can move from Draft after all labels are properly set. This process will be implemented using Github actions.

Feedback:

BL:This seems ok to me. We’ll need to define who can change the labels, and what labels should be used to trigger reviews and then implementation.

CR: What is the definition of a Draft PR please? We’re waiting on FINOS to give us Triage permissions on the Editor role which will allow contributors to add the labels themselves which should help here. We still need to cater for general Participants submitting PRs though so a Draft status would make sense. The more that can be done automatically the better.

Result: General agreement. Draft is defined by Github. Moving to Ready state would require proper labelling. Labelling is the responsibility of the contributor but maintainers may have an opinion on the labels.

**Comment by chrisisla** [comment]
**Result**:
- Draft is defined by Github. 
- Moving to Ready state would require proper labelling. 
- Labelling is the responsibility of the contributor but maintainers may have an opinion on the labels.

This looks good to me @tomhealey-icma 

One question for you @minesh-s-patel - when contributing a workspace from within Rosetta it currently creates a standard Pull Request. I'm assuming Rosetta can be updated to create a Draft Pull Request instead?

**Comment by llynhiavu** [comment]
Lyteck (ISDA): proposal makes sense.

**Comment by lolabeis** [comment]
Agree with the proposal. And in relation to the question:

> when contributing a workspace from within Rosetta it currently creates a standard Pull Request. I'm assuming Rosetta can be updated to create a Draft Pull Request instead?

Yes indeed we'll need to adjust

**Reply by chrisisla** [reply]
Awesome, thanks for confirming @lolabeis 🙂

**Comment by tomhealey-icma** [comment]
I'm looking to close this issue as the majority has approved. I've also submitted a PR, [2924](https://github.com/finos/common-domain-model/pull/2924) to automate the label and action.

**Reply by lolabeis** [reply]
Sounds good.

I suggest you transform this Discussion into an Issue (using the "Create issue from discussion" option). You will then be able to associate a PR to that Issue and Issue will automatically close once PR gets merged.
